### PR TITLE
feature/testnet-deployments

### PR DESCRIPTION
**Pull request by intl-man**
Automated deployments for each feature branch push.

### CODE REVIEW

**Review by ilchebedelovski — DISMISSED** [review]
LGTM

File: build/rakelib/cluster.rake [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace :cluster do
   task :destroy, [:chainnet, :provider] do |t, args|
     check_args(args)
     puts "Destroy running cluster: #{path(args)}"
-    system("cd #{path(args)} && terraform destroy") or exit 1
+    system("cd #{path(args)} && terraform destroy -auto-approve") or exit 1
```
Comment (utx0) [inline build/rakelib/cluster.rake:51]: I woundn't do `-auto-approve` on destroy...

File: build/terraform/template/aws/cluster.tf.tpl [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
-//        hostname = "app.terraform.io"
-//    }
-//}
+terraform {
+    backend "remote" {
+        hostname = "app.terraform.io"
+        organization = "Sifchain"
+
+        workspaces {
+            name = "sifchain-aws-feature-testnets"
```
Comment (utx0) [inline build/terraform/template/aws/cluster.tf.tpl:8]: Hey this should only be edited once the template has been scaffolded to `./live/*` Can we revert this.

File: build/terraform/template/aws/cluster.tf.tpl [hunk 3]
```diff
@@ -16,8 +21,8 @@
 
 // Sifchain terraform module
 module sifchain {
-    source                  = "../../build/terraform/providers/aws"
-    region                  = "us-west-1"
+    source                  = "github.com/sifchain/sifnode/build/terraform/providers/aws"
```
Comment (utx0) [inline build/terraform/template/aws/cluster.tf.tpl:24]: I would keep this relative as this means you have to have the repo cloned and you can also make edits locally and not change this unnecessarily

File: .github/workflows/feature.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        shell: bash
+        run: |
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+      - name: Login to Docker Hub
+        run: docker login -u ${{ secrets.DOCKERHUB_USERNAME }} -p ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build image
+        run: |
+          make CHAINNET=${{ steps.vars.outputs.sha_short }} BINARY=sifnoded build-image
```
Comment (utx0) [inline .github/workflows/feature.yml:30]: We might need to handle this differently with the work I am in the process of doing so we can create a testnet build with testnet code ie fauct. Let talk when you are online to work this out.

**Review by utx0 — CHANGES_REQUESTED** [review]

**Review by utx0 — DISMISSED** [review]

File: build/helm/sifnode/templates/sifnode-deployment.yaml [hunk 5]
```diff
@@ -20,8 +20,8 @@ spec:
       labels:
         {{- include "sifnode.selectorLabels" . | nindent 8 }}
     spec:
-      imagePullSecrets:
-        - name: {{ include "sifnode.fullname" . }}-ghcr
+#      imagePullSecrets:
+#        - name: {{ include "sifnode.fullname" . }}-ghcr
```
Comment (utx0) [inline build/helm/sifnode/templates/sifnode-deployment.yaml:24]: Clean up?

File: build/helm/sifnode/values.yaml [hunk 6]
```diff
@@ -36,6 +30,11 @@ securityContext: {}
   # runAsNonRoot: true
   # runAsUser: 1000
 
+image:
+  repository: sifchain/sifnoded
+  pullPolicy: IfNotPresent
+  tag: "develop"
```
Comment (utx0) [inline build/helm/sifnode/values.yaml:36]: is this correct?

**Review by utx0 — APPROVED** [review]

**Pull request merged**